src/zjs_spi.c

```diff
@@ -0,0 +1,314 @@
+// Copyright (c) 2017, Intel Corporation.
+
+// Zephyr includes
+#ifndef ZJS_LINUX_BUILD
+#include <zephyr.h>
+#include <spi.h>
+#include <misc/util.h>
+#endif
+#include <string.h>
+#include <stdlib.h>
+
+// ZJS includes
+#include "zjs_spi.h"
+#include "zjs_util.h"
+#include "zjs_buffer.h"
+#include "zjs_callbacks.h"
+#include "zjs_error.h"
+
+#define SPI_BUS "SPI_"
+#define MAX_SPI_BUS 127
+#define MAX_READ_BUFF 1000
+#define MAX_DIR_LEN 13
+
+static jerry_value_t zjs_spi_prototype;
+
+static void zjs_spi_callback_free(void *native)
+{
+    // requires: handle is the native pointer we registered with
+    //             jerry_set_object_native_handle
+    //  effects: frees the spi item
+    spi_handle_t *handle = (spi_handle_t *)native;
+    handle->closed = true;
```

Comment:
I do because after you free it, the object still exists in JS until garbage collection.  The user can try and use the SPI object again before this happens, need a way to make sure they don't.